src/lib/y2storage/clients/partitions_proposal.rb

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env ruby
+#
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2storage"
+require "y2storage/dialogs/actions_presenter"
+require "installation/proposal_client"
+
+Yast.import "Popup"
+
+module Y2Storage
+  module Clients
+    # Proposal client to show the list of storage actions
+    class PartitionsProposal < ::Installation::ProposalClient
+      include Yast::Logger
+
+      def self.state(storage_manager)
+        return if staging_revision == storage_manager.staging_revision
+
+        self.staging_revision = storage_manager.staging_revision
+
+        staging = storage_manager.y2storage_staging
+        actiongraph = staging ? staging.actiongraph : nil
+        self.actions_presenter = ActionsPresenter.new(actiongraph)
+      end
+
+      def initialize
+        create_proposal
+        self.class.state(storage_manager)
```

Comment:
Even more weird: `state` (that looks like a method to query something) is called to actually initialize a value, but `create_proposal`(the looks like a method that indeed creates something) in many cases will actually not create a proposal (if there is already one).